File: csharp-dotnet2/src/main/CsharpDotNet2/IO/Swagger/Model/V3StopDepartureRequestRouteDirection.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// 
  /// </summary>
  [DataContract]
  public class V3StopDepartureRequestRouteDirection {
    /// <summary>
    /// Identifier of route; values returned by Routes API - v3/routes
    /// </summary>
    /// <value>Identifier of route; values returned by Routes API - v3/routes</value>
    [DataMember(Name="route_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "route_id")]
    public string RouteId { get; set; }

    /// <summary>
    /// Direction of travel identifier; values returned by Directions API - v3/directions
    /// </summary>
    /// <value>Direction of travel identifier; values returned by Directions API - v3/directions</value>
    [DataMember(Name="direction_id", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_id")]
    public int? DirectionId { get; set; }

    /// <summary>
    /// Name of direction of travel; values returned by Directions API - v3/directions
    /// </summary>
    /// <value>Name of direction of travel; values returned by Directions API - v3/directions</value>
    [DataMember(Name="direction_name", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "direction_name")]
    public string DirectionName { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class V3StopDepartureRequestRouteDirection {\n");
      sb.Append("  RouteId: ").Append(RouteId).Append("\n");
      sb.Append("  DirectionId: ").Append(DirectionId).Append("\n");
      sb.Append("  DirectionName: ").Append(DirectionName).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: dart/lib/model/v3_stop_departure_request_route_direction.dart
part of swagger.api;

class V3StopDepartureRequestRouteDirection {
  /* Identifier of route; values returned by Routes API - v3/routes */
  String routeId = null;
/* Direction of travel identifier; values returned by Directions API - v3/directions */
  int directionId = null;
  //enum directionIdEnum {  0,  2147483647,  };
/* Name of direction of travel; values returned by Directions API - v3/directions */
  String directionName = null;

  V3StopDepartureRequestRouteDirection();

  @override
  String toString() {
    return 'V3StopDepartureRequestRouteDirection[routeId=$routeId, directionId=$directionId, directionName=$directionName, ]';
  }

  V3StopDepartureRequestRouteDirection.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    routeId = json['route_id'];
    directionId = json['direction_id'];
    directionName = json['direction_name'];
  }

  Map<String, dynamic> toJson() {
    return {
      'route_id': routeId,
      'direction_id': directionId,
      'direction_name': directionName
     };
  }

  static List<V3StopDepartureRequestRouteDirection> listFromJson(List<dynamic> json) {
    return json == null ? new List<V3StopDepartureRequestRouteDirection>() : json.map((value) => new V3StopDepartureRequestRouteDirection.fromJson(value)).toList();
  }

  static Map<String, V3StopDepartureRequestRouteDirection> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, V3StopDepartureRequestRouteDirection>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new V3StopDepartureRequestRouteDirection.fromJson(value));
    }
    return map;
  }
}
